PocketKit/Sources/Sync/Operations/DeleteItem.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+import Apollo
+import Combine
+
+
+class DeleteItem: AsyncOperation {
+    private let apollo: ApolloClientProtocol
+    private let events: PassthroughSubject<SyncEvent, Never>
+    private let itemID: String
+
+    init(
+        apollo: ApolloClientProtocol,
+        events: PassthroughSubject<SyncEvent, Never>,
+        itemID: String
```

Comment:
I feel like this is an initializer we'll see for… most? all? item mutations. Ideas:

1. Define a protocol
2. Subclass `AsyncOperation` to `AsyncItemMutation` or something, and then just have a subclass fill in `main`
3. None of the above!

I'm alright with any of the three.

Reply:
Agree. We are starting to see a lot of duplication here. I am working on the Archive action now and will be refactoring to eliminate it as part of that PR. Sound good?

Reply:
Absolutely!